package.json

```diff
@@ -15,7 +15,10 @@
     "react-bootstrap": "^1.0.1",
     "react-dom": "^16.13.1",
     "react-icons": "^3.10.0",
+    "react-redux": "^7.2.0",
     "react-scripts": "3.4.1",
+    "redux": "^4.0.5",
+    "redux-thunk": "^2.3.0",
```

Comment:
Thunk is a good first choice. Have you ever seen redux-observables? Let's discuss.